rsconnect/api.py

```diff
@@ -312,28 +368,57 @@ def drop_context(self):
         gc.collect()
         return self
 
-    def setup_connect_server(
+    def setup_remote_server(
         self,
         name: str = None,
         url: str = None,
         api_key: str = None,
         insecure: bool = False,
         cacert: IO = None,
         ca_data: str = None,
+        account: str = None,
+        token: str = None,
+        secret: str = None,
     ):
-        if name and url:
-            raise RSConnectException("You must specify only one of -n/--name or -s/--server, not both.")
-        if not name and not url:
-            raise RSConnectException("You must specify one of -n/--name or -s/--server.")
+        validation.validate_connection_options(
+            name=name,
+            url=url,
+            api_key=api_key,
+            insecure=insecure,
+            cacert=cacert,
+            account=account,
+            token=token,
+            secret=secret,
+        )
 
         if cacert and not ca_data:
             ca_data = text_type(cacert.read())
 
-        url, api_key, insecure, ca_data, _ = ServerStore().resolve(name, url, api_key, insecure, ca_data)
-        self.connect_server = RSConnectServer(url, api_key, insecure, ca_data)
+        server_data = ServerStore().resolve(name, url)
+        if server_data.from_store:
+            url = server_data.url
+            api_key = server_data.api_key
+            insecure = server_data.insecure
+            ca_data = server_data.ca_data
+            account = server_data.account
+            token = server_data.token
+            secret = server_data.secret
+        self.is_server_from_store = server_data.from_store
+
+        if api_key:
+            self.remote_server = RSConnectServer(url, api_key, insecure, ca_data)
+        elif token and secret:
+            self.remote_server = ShinyappsServer(url, account, token, secret)
+        else:
+            self.remote_server = ShinyappsServer(url, account, token, secret)
```

Comment:
This seems a little odd. If it's possible to pass in no arguments when creating a server (api_key, token, secret) why would it end up create a shinyapps server. 

I would have expected either a validation error - or a `RSConnectServer` for backwards compatibility?

Reply:
per Sean's comment, this should look as follows:
https://github.com/rstudio/rsconnect-python/blob/3e4b0a7ab31ba41cc5b070522be0cc4bfd2efc49/rsconnect/api.py#L410-L416